core/vm/zk_transaction_counters.go

```diff
@@ -18,16 +18,28 @@ type TransactionCounter struct {
 	smtLevels          int
 }
 
-func NewTransactionCounter(transaction types.Transaction, smtMaxLevel int) *TransactionCounter {
+func NewTransactionCounter(transaction types.Transaction, smtMaxLevel int, shouldCountersBeUnlimited bool) *TransactionCounter {
 	totalLevel := calculateSmtLevels(smtMaxLevel, 32)
-	tc := &TransactionCounter{
-		transaction:        transaction,
-		rlpCounters:        NewCounterCollector(totalLevel),
-		executionCounters:  NewCounterCollector(totalLevel),
-		processingCounters: NewCounterCollector(totalLevel),
-		smtLevels:          totalLevel,
-	}
 
+	var tc *TransactionCounter
+
+	if shouldCountersBeUnlimited {
+		tc = &TransactionCounter{
+			transaction:        transaction,
+			rlpCounters:        NewUnlimitedCounterCollector(),
+			executionCounters:  NewUnlimitedCounterCollector(),
+			processingCounters: NewUnlimitedCounterCollector(),
+			smtLevels:          256, // max depth of the tree anyways
```

Comment:
We should make this number as small as possible, the larger it is the more counters will be consumed, so setting this to 1 will help get us use less counters.